### ISSUE TITLE
AVPro 1.11.7 Update failes to import/install

### ISSUE THREAD

**Issue by jules43** [RenderHeads/UnityPlugin-AVProVideo#706]
**Describe the bug**
I have a blank 3D Unity project. I opened the package manager which shows 1.11.3 as currently downloaded and 1.11.7 available for update. I hit the update button and it starts downloading however when it completes the status is unchanged.

I import the 1.11.3 version using the import button, it installs successfully. I still can't update the package. Each time I hit the import button the progress bar finishes and the outcome remains the same.

**Your Setup (please complete the following information):**
 - Unity version: 2020.3.2f1 Personal
 - AVPro Video version: 1.11.3 / 7
 - Operating system version: Windows 10
 - Device model: Desktop PC
 - Video specs (resolution, frame-rate, codec, file size): na

**To Reproduce**
1. Create a new 3D unity project
2. Open the packet manager 
3. Import 1.11.3 of AV Pro
4. Attempt to update to 1.11.7

**Logs**
If applicable, add error logs to help explain your problem.

**Screenshots**

**Videos**
If applicable, add a copy of your video or the URL

### Please DO NOT LINK / ATTACH YOUR PROJECT FILES HERE
Instead email the link to us unitysupport@renderheads.com

**Comment by kahnivore** [comment]
Hi @jules43 

Have you tried closing Unity and trying to import immediately after opening? It's possible that the dll files are locked and preventing you from replacing them.

**Comment by jules43** [comment]
The behaviour is the same after shutting down Unity, reloading and attempting to import/update again. I will try rebooting the machine to make sure there's no other state hanging about.

When the package manager completes the Update, the UI briefly shows it as being 1.11.7 but then it returns to showing it as 1.11.3.

The 1.11.3 version of AVPro samples don't all build in 2020.3 so I will try the same sequence of steps with 2019.4.23 and see if it behaves differently.

(This issue is not currently urgent for me personally as we will be upgrading to V2 and have a working build with an older version of AVPro)

**Comment by kahnivore** [comment]
Thanks. Will be good to hear if the update works in 2019.x.

**Comment by jules43** [comment]
tl;dr Problem seems to be a bug in Unity with the way it handles download/updating in the PackageManager when you already have an older version of the plug-in.

The 2019 version displays an error message whereas the 2020 version fails invisibly.

The workaround for me was to go to the directory where the PackageManager downloads unity packages and delete the AVPro ones.

...\AppData\Roaming\Unity\Asset Store-5.x\RenderHeads Ltd\ScriptingVideo

Then load Unity / Package Manager and download AVPro, ignore the error, quit unity, reload Unity and do the import.


Longer version of story:

I did a fresh reboot and then downloaded the Unity 2019.4.23f1 version of Unity, and used the most recent version of Unity Hub to create a new 3D project based on this.

I opened the Window->Package Manager and selected My Assets from the filter. This showed AVPro Video 1.x in the list (both 11.3 and 11.7 versions).

I hit the update button and this time I got an error message (see image).

Basically it's complaining about being unable to open the unitypackage it has just decrypted.

So I went to the location specified and found three files:
   AVPro Video 1x-content__56355.tmp
   AVPro Video 1x.unitypackage
   AVPro Video.unitypackage

As a test I tried renaming the 1.x unitypackage file and it was locked, so I shut down unity and unity hub and then renamed the unitypackage files, then reloaded the test project and reopened the package manager.

AVPro was listed as before except the import/update buttons now showed download (and only the 11.7 version).

I hit download and it seemed to download and start decrypting before coming up with the same error.

Checking back in the directory the AVPro 1.x files were both back (as expected). The tmp file unlocked and the unitypackage locked.

So I shutdown and reloaded the project again and re-opened the Package Manager. This time the 1.11.7 version showed up in My Assets as available for import and I was able to successfully import the package with no errors.

I then opened the test project in Unity 2020.4.2f1 and accepted the offer to upgrade.

It's worth noting that the upgrade made important changes to the AVPro code due to some function renaming between 2019 and 2020 versions.

A final note is that even with the upgrade handling MediaPlayer.cs:1024 gives a warning about isNetworkError being deprecated.

**Comment by kahnivore** [comment]
Closing this because it was resolved by user.
